Crawlers/news_sites/middlewares.py

```diff
@@ -20,30 +20,30 @@ def from_crawler(cls, crawler):
         crawler.signals.connect(s.spider_opened, signal=signals.spider_opened)
         return s
 
-    def process_spider_input(response, spider):
+    def process_spider_input(self, response, spider):
         # Called for each response that goes through the spider
         # middleware and into the spider.
 
         # Should return None or raise an exception.
         return None
 
-    def process_spider_output(response, result, spider):
+    def process_spider_output(self, response, result, spider):
         # Called with the results returned from the Spider, after
         # it has processed the response.
 
         # Must return an iterable of Request, dict or Item objects.
         for i in result:
             yield i
 
-    def process_spider_exception(response, exception, spider):
+    def process_spider_exception(self, response, exception, spider):
```

Comment:
@Anmolbansal1 Is there a reason for binding these functions with classes?

Reply:
This is instantiated by default with `scrapy startproject` command and I think it wouldn't be good to change it